### PR TITLE
add Langflow

### PR DESCRIPTION
**Pull request by ZacharyKehlGEAppliances**
## Key Features:

- Full integration with Harbor's container management
- SQLite database by default with optional PostgreSQL support
- Preconfigured authentication and default admin credentials
- Compatible with Harbor's standard service commands

## Configuration Details:

- Default port: `7860`
- Admin login: `admin@admin.com`
- Admin password: `admin`
- Data persistence in Harbor's standard data directory

### Example Usage:
```
harbor up langflow
```

### With PostgreSQL database
```
harbor up langflow langflow-db
```
### Access UI
```
harbor open langflow
```

## Technical Implementation

- Uses official langflowai/langflow image
- Integrates with Harbor's network infrastructure
- Follows Harbor's service conventions for configuration
- Provides standard Harbor CLI management commands

## Changes made

- Added core compose files for Langflow service
- Added cross-service integration files for database support
- Added environment variable definitions
- Updated service documentation and help screens

## Testing

- Verified service startup and persistence
- Tested database integration
- Confirmed UI accessibility
- Validated Harbor command integration

### CODE REVIEW

**Comment by ZacharyKehlGEAppliances** [comment]
Looks like I need to update README and Wiki

**Comment by av** [comment]
This is awesome, thank you so much for the integration! 💯 

 I'll take a closer look in the next few days. The docs/wiki - I'm quite sure there's no easy way to contribute to them externally, so I'll write them when testing the integration, similar to the previous contributions (example: https://github.com/av/harbor/wiki/2.3.26-Satellite:-Qdrant)

Thanks again!

File: harbor.sh [hunk 1]
```diff
@@ -3505,6 +3506,96 @@ run_boost_command() {
     esac
 }
 
+run_langflow_command() {
```
Comment (av) [inline harbor.sh:3509]: Very nice, I was lazy to write these for a lot of recent services, kudos!

File: compose.x.langflow.postgres.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+services:
+  langflow:
+    environment:
+      - LANGFLOW_DATABASE_URL=postgresql://${HARBOR_LANGFLOW_DB_USER:-postgres}:${HARBOR_LANGFLOW_DB_PASSWORD:-password}@${HARBOR_CONTAINER_PREFIX}.langflow-db:5432/${HARBOR_LANGFLOW_DB_NAME:-langflow}
```
Comment (av) [inline compose.x.langflow.postgres.yml:4]: Probably a leftover file

File: compose.langflow.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    image: langflowai/langflow:${HARBOR_LANGFLOW_VERSION:-latest}
+    ports:
+      - "${HARBOR_LANGFLOW_HOST_PORT:-7860}:7860"
+    volumes:
+      - ${HARBOR_LANGFLOW_DATA:-/data/harbor/langflow}:/var/lib/langflow
+      - ./langflow/configs:/app/configs
+    env_file:
+      - ./.env
+      - ./langflow/override.env
+    environment:
```
Comment (av) [inline compose.langflow.yml:13]: Everything in the override.env can be removed here - will make it easier to adjust via `harbor env`

File: compose.langflow.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+services:
+  langflow:
+    container_name: ${HARBOR_CONTAINER_PREFIX:-harbor}.langflow
+    image: langflowai/langflow:${HARBOR_LANGFLOW_VERSION:-latest}
+    ports:
+      - "${HARBOR_LANGFLOW_HOST_PORT:-7860}:7860"
```
Comment (av) [inline compose.langflow.yml:6]: The default values/env are all defined in the profiles/default.env, it gets merged into .env during install, and updates (use `harbor config update` to merge in-dev)

File: langflow/harbor.prompt [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+# langflow/harbor.prompt
+# This is the prompt file for Langflow integration
+
+__HARBOR_LANGFLOW_DESCRIPTION="Langflow: A LangChain Flow UI"
+__HARBOR_LANGFLOW_DOCS="https://docs.langflow.org"
+__HARBOR_LANGFLOW_MAIN_PORT=7860
+__HARBOR_LANGFLOW_HELP="Langflow is a UI for LangChain, designed to help you prototype and build LLM applications."
+
+main() {
+    return 0
```
Comment (av) [inline langflow/harbor.prompt:11]: This file is probably also a leftover

File: compose.x.langflow.postgres.yml (hunk 2 above)
Comment (ZacharyKehlGEAppliances) [reply compose.x.langflow.postgres.yml:4]: Do these need to be removed?

Comment (av) [reply compose.x.langflow.postgres.yml:4]: Yes, unused files can confuse someone else in the future, so better to only keep the relevant bits

**Comment by ZacharyKehlGEAppliances** [comment]
Updated with _hopefully_ the correct changes!

**Review by av — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**